contextily/place.py

```diff
@@ -8,6 +8,11 @@
 from . import providers
 from ._providers import TileProvider
 
+# Set user ID for Nominatim
+rng = np.random.default_rng()
+val = rng.integers(1000000)
+gp.geocoders.options.default_user_agent = f"contextily_user_{val}"
```

Comment:
Can we only set this in the actual geocoder object we create below? 
Because now importing contextily will have a side effect on the geopy module

Reply:
@darribas are you OK with only passing this "contextily_user_val" to the `gp.geocoders.Nominatim()` default keyword value?

Reply:
@darribas are you ok with this? Then I can push a change, merge this, and do a small bug fix release

Reply:
This sounds great and a much cleaner way to go about. I'm +1 on this